projects/components/src/filtering/filter/filter-delimiters.ts

```diff
@@ -1,3 +1,11 @@
+import { isNonEmptyString } from '@hypertrace/common';
+
 export const ARRAY_DELIMITER = ',';
 export const MAP_LHS_DELIMITER = '.';
 export const MAP_RHS_DELIMITER = ':';
+
+export const splitFirstOccurrence = (str: string, delimiter: string): string[] => {
+  const firstIndex = str.indexOf(delimiter);
+
+  return [str.substr(0, firstIndex), str.substr(firstIndex + 1)].filter(isNonEmptyString);
```

Comment:
nit - I don't think this handles edge cases correctly (at least anything where a empty string is part of the expected split result such as a string that starts or ends with the delimiter, or an empty string)

Reply:
Added a test for that case.

Reply:
the test cases look fine, but doesn't this still behave differently than split itself? If that's acceptable for how it's used then it doesn't need to change, but should be local to the file it's used vs exported like a generic util.

Reply:
Just noticed the name changed in this rev - that's better, but I think it'd be cleaner to just keep it local